src/components/TxHistory.tsx

```diff
@@ -0,0 +1,338 @@
+    const endIndex = startIndex + itemsPerPage;
+    return dataTest?.slice(startIndex, endIndex); // ! change dataTest to data
+  }, [currentPage, dataTest]); // ! change dataTest to data
+
+  const handleNextPage = () => {
+    if (currentPage * itemsPerPage < (dataTest?.length ?? 0)) {
+      // ! change dataTest to data
+      setCurrentPage(currentPage + 1);
+    }
+  };
+
+  const handlePreviousPage = () => {
+    if (currentPage > 1) {
+      setCurrentPage(currentPage - 1);
+    }
+  };
+
+  const handleOpenLink = (txid: string) => {
+    window.open(`${URL_WHATSINCHAIN}${txid}`, '_blank');
+  };
+
+  const getIconForSummary = (tag: string, icon: string) => {
+    switch (tag) {
+      case 'fund':
+        return bsvCoin;
+      case 'lock':
+        return lock;
+      case 'list':
+        return priceTag;
+      default:
+        return icon
+          ? `${gorillaPoolService.getBaseUrl(chromeStorageService.getNetwork())}/content/${icon}`
+          : GENERIC_TOKEN_ICON;
+    }
+  };
+
+  const toggleRowExpansion = (idx: number) => {
+    setExpandedRows((prev) => {
+      const newSet = new Set(prev);
+      if (newSet.has(idx)) newSet.delete(idx);
+      else newSet.add(idx);
+      return newSet;
+    });
+  };
+
+  return (
+    <Container theme={theme} isSlidingOut={isSlidingOut}>
+      <BackWrapper>
+        <FaTimes size={'1.5rem'} color={theme.color.global.contrast} cursor="pointer" onClick={handleBackClick} />
+      </BackWrapper>
+      <Text style={{ marginTop: '3rem', fontSize: '1.25rem', fontWeight: 700 }} theme={theme}>
+        See Last Activity
+      </Text>
+      {(paginatedData || []).length > 0 ? (
+        paginatedData?.map((t) => {
+          const summaryEntries = Object.entries(t.summary);
+          const isExpanded = expandedRows.has(t.idx);
+
+          return (
+            <RowWrapper>
+              <FavoriteRow
+                theme={theme}
+                key={t.idx}
+                onClick={summaryEntries.length > 1 ? () => toggleRowExpansion(t.idx) : undefined}
+                style={
+                  summaryEntries.length > 1
+                    ? {
+                        display: 'flex',
+                        alignItems: 'center',
+                        cursor: 'pointer',
+                        color: theme.color.global.gray,
+                      }
+                    : {
+                        display: 'flex',
+                        alignItems: 'center',
+                        color: theme.color.global.gray,
+                      }
+                }
+              >
+                <TickerWrapper
+                  style={{
+                    width: '100%',
+                    display: 'flex',
+                    flexDirection: 'column',
+                    gap: '0.5rem',
+                  }}
+                >
+                  {summaryEntries.slice(0, isExpanded ? summaryEntries.length : 1).map(([key, value], idx) => (
+                    <div
```

Comment:
Use of `div` throughout...

Use styled-components and give these divs a more explicit name to match the existing pattern throughout the code base.